compiler/src/iree/compiler/Codegen/Common/TopkLowering.cpp

```diff
@@ -0,0 +1,915 @@
+                Value newInds = b.create<tensor::InsertOp>(
+                    loc, shiftElemsLoop.getRegionIterArgs()[3],
+                    shiftElemsLoop.getRegionIterArgs()[1],
+                    ValueRange{idx0, nextIndVar});
+                SmallVector<Value> newOperands;
+                newOperands.push_back(newVals);
+                newOperands.push_back(newInds);
+                newOperands.push_back(out1Repl);
+                newOperands.push_back(out2Repl);
+                b.create<scf::YieldOp>(loc, newOperands);
+              }
+
+              lastLoopOp.erase();
+
+              // Get the smellest element as the last added element
+              // to return to the next iteration of the tile (1x16xf/i32) loop.
+              smElemElse = b.create<tensor::ExtractOp>(
+                  loc, shiftElemsLoop.getResult(0),
+                  ValueRange{idx0, ifExpand.getResult(0)});
+              addElemsElse =
+                  b.create<arith::AddIOp>(loc, ifExpand.getResult(0), one);
+              outValsElse = shiftElemsLoop.getResult(0);
+              outIndsElse = shiftElemsLoop.getResult(1);
+
+              // Insert the element in the output.
+              outValsElse = b.create<tensor::InsertOp>(
+                  loc, elemToInsertIfNeeded, outValsElse,
+                  ValueRange{idx0, insertionIndex});
+              outIndsElse = b.create<tensor::InsertOp>(
+                  loc,
+                  b.create<arith::IndexCastOp>(loc, b.getIntegerType(32),
+                                               elemIndex),
+                  outIndsElse, ValueRange{idx0, insertionIndex});
+              SmallVector<Value> newOperandsElse;
+              newOperandsElse.push_back(outValsElse);
+              newOperandsElse.push_back(outIndsElse);
+              newOperandsElse.push_back(smElemElse);
+              newOperandsElse.push_back(addElemsElse);
+              b.create<scf::YieldOp>(loc, newOperandsElse);
+            });
+
+        // Propagate returns from nested loops and ifs to the
+        // output of the function.
+        Value outValsAtEnd = ifAddAtEnd.getResult(0);
+        Value outIndsAtEnd = ifAddAtEnd.getResult(1);
+        Value smElemAtEnd = ifAddAtEnd.getResult(2);
+        Value addElemsAtEnd = ifAddAtEnd.getResult(3);
+
+        SmallVector<Value> newOperands;
+        newOperands.push_back(outValsAtEnd);
+        newOperands.push_back(outIndsAtEnd);
+        newOperands.push_back(smElemAtEnd);
+        newOperands.push_back(addElemsAtEnd);
+        b.create<scf::YieldOp>(loc, newOperands);
+      });
+  // Build and return the continuation of the outer loop,
+  // after inserting the element.
+  return InsertElemContinuationValues{
+      ifAddElem.getResult(0), ifAddElem.getResult(1), ifAddElem.getResult(2),
+      ifAddElem.getResult(3)};
+}
+
+/// Vectorize a `topKOp` with (1) static result and input types
+static LogicalResult lowerAsLinalgExtTopkUsingSCF(
```

Comment:
Instead of return by reference, make this return a `FailureOr<SmallVector<Value>>`

Reply:
Changed! Thanks!